packages/core-mobile/app/new/features/portfolio/hooks/useBalanceInCurrencyForAccount.ts

```diff
@@ -20,26 +19,19 @@ export const useBalanceInCurrencyForAccount = (
   isLoadingBalance: boolean
   balance: number
 } => {
-  const activeAccount = useSelector(selectActiveAccount)
   const activeWallet = useSelector(selectActiveWallet)
   const account = useSelector(selectAccountById(accountId))
 
-  const isActiveAccount =
-    activeAccount && account && activeAccount.id === account.id
-
-  const { data: accountBalance, isLoading: isLoadingAccountBalances } =
-    useAccountBalances(isActiveAccount ? activeAccount : undefined)
-
   const { data: walletBalances } = useWalletBalances(activeWallet)
 
   const balanceTotalInCurrency = useBalanceTotalInCurrencyForAccount({
     account,
-    sourceData: isActiveAccount ? accountBalance : walletBalances?.[accountId]
+    // TODO: fix type mismatch after fully migrating to the new backend balance types
+    // @ts-ignore
+    sourceData: walletBalances[accountId]
```

Comment:
there is no need to conditionally check active account and pass in the correct data source anymore since `useWalletBalances` is now just a series of `useAccountBalances` (they share the same cache key). thus, we can always just use `useWalletBalances` here.